test/ConsolueModuleLoader.test.cpp

```diff
@@ -66,3 +66,34 @@ TEST(ConsoleModuleLoader, ModuleWithEmptyDLL)
 	EXPECT_NO_THROW(consoleModuleLoader.RunModules());
 	EXPECT_NO_THROW(consoleModuleLoader.UnloadModules());
 }
+
+TEST(ConsoleModuleLoader, MultiModule) {
+	const auto exeDir = fs::path(GetGameDirectory());
+	const std::vector<std::string> moduleNames{"Module1", "Module2"};
+
+	// Create some empty test module directories
+	for (const auto& moduleName : moduleNames) {
+		fs::create_directory(exeDir / moduleName);
+	}
+
+	ConsoleModuleLoader consoleModuleLoader(moduleNames);
+
+	EXPECT_EQ(2u, consoleModuleLoader.Count());
+	EXPECT_EQ(moduleNames[0], consoleModuleLoader.GetModuleName(0));
+	EXPECT_EQ(moduleNames[1], consoleModuleLoader.GetModuleName(1));
+
+	EXPECT_TRUE(consoleModuleLoader.IsModuleLoaded(moduleNames[0]));
+	EXPECT_TRUE(consoleModuleLoader.IsModuleLoaded(moduleNames[1]));
+
+	EXPECT_FALSE(consoleModuleLoader.IsModuleLoaded(""));
+
+	EXPECT_NO_THROW(consoleModuleLoader.LoadModules());
+	EXPECT_NO_THROW(consoleModuleLoader.RunModules());
+	EXPECT_NO_THROW(consoleModuleLoader.UnloadModules());
+
+	// // Cleanup test module directories
+	// for (const auto& moduleName : moduleNames) {
```

Comment:
I think I would prefer this be un-commented and guarded by a preprocessor directive to prevent it running on mingw but still let it cleanup on MSVC builds.

Reply:
I'm hesitant to jump straight to a compiler/platform detect solution. I'd like to experiment a bit more to see if I can get something working across all compilers. Plus, it would be nice to cleanup all of the test directory creation.